chain/types/ethtypes/eth_transactions.go

```diff
@@ -122,6 +125,12 @@ func EthTxArgsFromUnsignedEthMessage(msg *types.Message) (EthTxArgs, error) {
 		}
 		to = &addr
 
+		fromAddr, err := EthAddressFromFilecoinAddress(msg.From)
```

Comment:
There's a slight issue with this -- we might not be returning the "best" possible EthAddress here. In particular, we'd like to resolve to _actual_ eth addresses if the `msg.From` is an ID address (by looking up the corresponding f4 address). See `node/impl/full/eth.go`'s `lookupEthAddress` for the best-effort implementation of this.

I think we want to make this change at the API level when forming an `EthTx`. I think `newEthTxFromSignedMessage` is likely the best place for this.